include/string.h

```diff
@@ -35,8 +35,16 @@
  ****************************************************************************/
 
 #define strcoll_l(s1, s2, l)    strcoll(s1, s2)
+#define strdupa(x)              strcpy(alloca(strlen(x) + 1), x)
 #define strerror_l(e, l)        strerror(e)
 #define strxfrm_l(s1, s2, n, l) strxfrm(s1, s2, n)
+#define	strndupa(x, len) ({ \
```

Comment:
({ is gcc extension, let's change to c function?

Reply:
Done!

Reply:
keep, strndupa return alloca local memory, so using marco to do it.

Reply:
Ok